### ISSUE TITLE
OOP vs FP in practice

### ISSUE THREAD

**Issue by bouwe77** [bouwe77/curious-coders#16]
I've done OOP (C#) for years, and am now super happy with more of a FP kinda approach, certainly no classes anymore.

Only using classes is not by definition OOP, while not using classes, but only functions, is not FP of course.

However, what is the practical difference between the two? Of course design patterns, also state comes into mind, and to me personally FP feels to me like less boilerplate.

Perhaps we could build a very simple piece of functionality in both OOP and FP and find what is different?

Only to find out they do not exclude each other of course. But is it really a matter of preference, or is there more?

**Comment by RichStone** [comment]
FP for me is a bit like black magic. A PARADIGM. Either you start learning it some day, or stick with what you know and learned so far (which most of the time is OOPish). Looking forward to get this a bit more clarified ;))

**Comment by bouwe77** [comment]
Yeah definitely can relate. What helped for me was moving to frontend and JS/TS in general (also backend), where a more functional approach is more common.
